tests/unit/test_row_noise.py

```diff
@@ -45,12 +46,70 @@ def test_omission(dummy_data):
     assert set(noised_data1.columns) == set(dummy_data.columns)
     assert (noised_data1.dtypes == dummy_data.dtypes).all()
 
+
+def test_do_not_respond(mocker, dummy_data):
+    config = get_configuration()[DATASETS.census.name][Keys.ROW_NOISE][
+        NOISE_TYPES.do_not_respond.name
+    ]
+    mocker.patch(
+        "pseudopeople.noise_functions._get_census_omission_noise_levels",
+        side_effect=(lambda *_: config[Keys.ROW_PROBABILITY]),
+    )
+    dataset_name_1 = DATASETS.census.name
+    dataset_name_2 = DATASETS.acs.name
+    my_dummy_data = dummy_data.copy()
+    my_dummy_data["age"] = 27
+    my_dummy_data["sex"] = "Female"
+    my_dummy_data["race_ethnicity"] = "Vulcan"
+    noised_data1 = NOISE_TYPES.do_not_respond(
+        dataset_name_1, my_dummy_data, config, RANDOMNESS
+    )
+    noised_data2 = NOISE_TYPES.do_not_respond(
+        dataset_name_2, my_dummy_data, config, RANDOMNESS
+    )
+
+    # Test that noising affects expected proportion with expected types
+    assert np.isclose(
+        1 - len(noised_data1) / len(my_dummy_data), config[Keys.ROW_PROBABILITY], rtol=0.02
+    )
+    assert set(noised_data1.columns) == set(my_dummy_data.columns)
+    assert (noised_data1.dtypes == my_dummy_data.dtypes).all()
+
     # Check ACS data is scaled properly due to oversampling
-    expected_noise_2 = 0.5 + config[Keys.ROW_PROBABILITY] / 2
-    assert np.isclose(1 - len(noised_data2) / len(dummy_data), expected_noise_2, rtol=0.02)
-    assert set(noised_data2.columns) == set(dummy_data.columns)
-    assert (noised_data2.dtypes == dummy_data.dtypes).all()
+    expected_noise = 0.5 + config[Keys.ROW_PROBABILITY] / 2
+    assert np.isclose(1 - len(noised_data2) / len(my_dummy_data), expected_noise, rtol=0.02)
+    assert set(noised_data2.columns) == set(my_dummy_data.columns)
+    assert (noised_data2.dtypes == my_dummy_data.dtypes).all()
     assert len(noised_data1) != len(noised_data2)
+    assert True
+
+
+@pytest.mark.parametrize(
+    "age, race_ethnicity, sex, expected_level",
+    [
+        (3, "White", "Female", 0.0091),
+        (35, "Black", "Male", 0.0611),
+        (55, "Asian", "Female", 0),
+    ],
+)
+def test__get_census_omission_noise_levels(age, race_ethnicity, sex, expected_level):
+    """Test helper function for do_not_respond noising based on demography of age, race/ethnicity, and sex"""
+    pop = pd.DataFrame(
+        [[age, race_ethnicity, sex] for i in range(10)],
+        index=range(10),
+        columns=["age", "race_ethnicity", "sex"],
+    )
+    result = _get_census_omission_noise_levels(pop)
+    assert (np.isclose(result, expected_level, rtol=0.01)).all()
```

Comment:
Are you doing this because of float equality issues? Because mathematically these values should be equal as we have no randomness involved in this step. If it's sure to float equality issues, the `rtol` should be able to be much smaller.

Reply:
Yes, float equality. 0.91 vs 0.90, iirc. Much smaller? This is currently the smallest `rtol` in the repo.